cylc/flow/pathutil.py

```diff
@@ -153,30 +153,31 @@ def make_localhost_symlinks(rund, named_sub_dir):
                 f'Unable to create symlink to {src}.'
                 f' \'{value}\' contains an invalid environment variable.'
                 ' Please check configuration.')
-        make_symlink(src, dst)
+        symlink_success = make_symlink(src, dst)
         # symlink info returned for logging purposes, symlinks created
         # before logs as this dir may be a symlink.
-        symlinks_created[src] = dst
+        if symlink_success:
+            symlinks_created[src] = dst
     return symlinks_created
 
 
 def get_dirs_to_symlink(install_target, flow_name):
-    """Returns dictionary of directories to symlink from glbcfg."""
+    """Returns dictionary of directories to symlink from glbcfg.
+       Note the paths should remain unexpanded, to be expanded on the remote.
+    """
     dirs_to_symlink = {}
     symlink_conf = glbl_cfg().get(['symlink dirs'])
 
     if install_target not in symlink_conf.keys():
         return dirs_to_symlink
     base_dir = symlink_conf[install_target]['run']
     if base_dir is not None:
-        dirs_to_symlink['run'] = os.path.join(
-            expand_path(base_dir), 'cylc-run', flow_name)
+        dirs_to_symlink['run'] = os.path.join(base_dir, 'cylc-run', flow_name)
     for dir_ in ['log', 'share', 'share/cycle', 'work']:
         link = symlink_conf[install_target][dir_]
         if link is None or link == base_dir:
             continue
-        dirs_to_symlink[dir_] = os.path.join(
-            expand_path(link), 'cylc-run', flow_name, dir_)
+        dirs_to_symlink[dir_] = os.path.join(link, 'cylc-run', flow_name, dir_)
```

Comment:
Does this work for named/numbered run dirs?

Reply:
The `flow_name` should be the full named run.

Reply:
This is what I was concerned about, in the `register()` function: https://github.com/cylc/cylc-flow/blob/fa248cd3d21106a4b9792020718e4991390da93d/cylc/flow/workflow_files.py#L560-L561

Reply:
I've traced through this and as far as I can see, the register function has to be called with the full-named run. e.g. `myflow/run1`, I think the variable name `flow_name` perhaps would be better named something like `named_run` for clarity. So, I think the code works as expected in this function, I may be missing something though?

Reply:
Yes, I think it should be renamed. However, `named_run` would be `run1` or whatever. Really, it's the run dir, but as a path relative to `~/cylc-run`. So maybe `rel_run_dir`?

Reply:
As, below, leaving the renaming for universal id work.